rllib/core/torch/torch_sarl_trainer.py

```diff
@@ -0,0 +1,57 @@
+import torch
+from torch.nn.parallel import DistributedDataParallel as DDP
+
+import ray
+
+from ray.rllib.core.sarl_trainer import SARLTrainer
+from ray.rllib.utils.torch_utils import convert_to_torch_tensor
+
+
+class TorchSARLTrainer(SARLTrainer):
+    def __init__(self, config):
+        super().__init__(config)
+        gpu_ids = ray.get_gpu_ids()
```

Comment:
Just a note: this call will initialise a CUDA context that will break any downstream environments that need a GPU context (like visual simulators). This is a known CUDA bug that nobody seems to want to fix. `ray` will set `CUDA_VISIBLE_DEVICES` to the correct values which we can use without initializing a context.